### PR TITLE
TVF versions of the pull request and pull request events views to allow for more efficient querying.

### PR DESCRIPTION
**Pull request by pdewilde**

### CODE REVIEW

File: terraform/bigquery.tf [hunk 1]
```diff
@@ -505,6 +505,7 @@ module "metrics_views" {
 
   dataset_id    = google_bigquery_dataset.default.dataset_id
   base_table_id = google_bigquery_table.unique_events_view.table_id
+  base_tvf_id   = google_bigquery_routine.unique_events_by_date_type.routine_id
```
Comment (gjonathanhong) [inline terraform/bigquery.tf:508]: This should be either `pull_request_events_by_date` or `pull_requests_by_date`.

Comment (gjonathanhong) [inline terraform/bigquery.tf:508]:
```suggestion
  base_tvf_id   = google_bigquery_routine.pull_request_events_by_date.routine_id
```

This should be either `pull_request_events_by_date` or `pull_requests_by_date`.

File: terraform/modules/bigquery_metrics_views/variables.tf [hunk 2]
```diff
@@ -26,3 +26,8 @@ variable "base_table_id" {
   type        = string
   description = "The BigQuery base table name for creating metrics views from."
 }
+
+variable "base_tvf_id" {
+  type        = string
+  description = "The BigQuery base table name for creating metrics views from."
```
Comment (gjonathanhong) [inline terraform/modules/bigquery_metrics_views/variables.tf:32]: Fix description

File: terraform/modules/bigquery_metrics_views/tvf.tf [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    data_type = jsonencode({ typeKind : "TIMESTAMP" })
+  }
+}
+
+# Version of pull_requests that uses TVF
+resource "google_bigquery_routine" "pull_requests_by_date" {
+  project = var.project_id
+
+  dataset_id   = var.dataset_id
+  routine_id   = "unique_events_by_date_type"
```
Comment (gjonathanhong) [inline terraform/modules/bigquery_metrics_views/tvf.tf:49]: This seems unusual. Can two bigquery routines share the same ID?

Comment (pdewilde) [reply terraform/modules/bigquery_metrics_views/tvf.tf:49]: No, I'm just very bad at copy-pasta

**Pull request merged**